tests/test_tlsfuzzer_expect.py

```diff
@@ -1285,52 +1285,142 @@ def test_process_with_tls13_unallowed_extension(self):
+        state.msg_sock = mock.MagicMock()
+        ext = []
+        ext.append(SrvSupportedVersionsExtension().create((3, 4)))
+
+        msg = ServerHello().create(version=(3, 3),
+                                   random=bytearray(32),
+                                   session_id=bytearray(0),
+                                   cipher_suite=
+                                   CipherSuite.TLS_AES_128_GCM_SHA256,
+                                   extensions=ext)
 
         self.assertTrue(exp.is_match(msg))
 
         exp.process(state, msg)
 
-    def test_process_with_tls_1_3_wrong_downgrade_protection(self):
-        exp = ExpectServerHello(extensions={ExtensionType.renegotiation_info:
-                                            None},
-                                version=(3, 4), server_max_protocol=(3, 4))
+    def test_process_with_default_settings_and_tls_1_3_reply_with_1_2_downgrade_sentinel(self):
+        # check that if the server reply is obviously bogus (like when TLS 1.3
+        # ServerHello indicates that we are downgrading to TLS 1.2), the
+        # ServerHello is rejected
+        exp = ExpectServerHello()
 
         state = ConnectionState()
         client_hello = ClientHello()
-        ciph = CipherSuite.TLS_EMPTY_RENEGOTIATION_INFO_SCSV
-        client_hello.cipher_suites = [4, ciph]
+        client_hello.extensions = []
+        client_hello.cipher_suites = [CipherSuite.TLS_AES_128_GCM_SHA256]
+        ext = SupportedGroupsExtension().create([GroupName.secp256r1])
+        client_hello.extensions.append(ext)
+        c_ks = key_share_gen(GroupName.secp256r1)
+        ext = ClientKeyShareExtension().create([c_ks])
+        client_hello.extensions.append(ext)
+        ext = SupportedVersionsExtension().create([(3, 3), (3, 4)])
+        client_hello.extensions.append(ext)
         state.handshake_messages.append(client_hello)
         state.msg_sock = mock.MagicMock()
 
         rndbuf=bytearray(32)
         rndbuf[-8:] = TLS_1_2_DOWNGRADE_SENTINEL
+        ext = []
+        ext.append(SrvSupportedVersionsExtension().create((3, 4)))
 
-        msg = ServerHello().create(version=(3, 4),
+        msg = ServerHello().create(version=(3, 3),
                                    random=rndbuf,
                                    session_id=bytearray(0),
-                                   cipher_suite=4,
-                                   extensions=None)
+                                   cipher_suite=
+                                   CipherSuite.TLS_AES_128_GCM_SHA256,
+                                   extensions=ext)
 
         self.assertTrue(exp.is_match(msg))
 
-        with self.assertRaises(AssertionError):
+        with self.assertRaises(AssertionError) as e:
             exp.process(state, msg)
 
+        self.assertIn("downgrade protection sentinel but shouldn't",
+                      str(e.exception))
+
+    def test_process_with_default_settings_and_tls_1_3_reply_with_1_1_downgrade_sentinel(self):
```

Comment:
as above I would shorten this to test_process_with_incorrect_tls_1_1_downgrade_sentinel()